tracer/test/test-applications/integrations/Samples.DataStreams.Kafka/Program.cs

```diff
@@ -211,21 +211,17 @@ void HandleFanIn(ConsumeResult<string, string> consumeResult)
         _fanInMessages.Add(consumeResult);
         if (_fanInMessages.Count == 3)
         {
-            Stack<IDisposable> scopes = new();
+            var iteration = 1;
             foreach (var fanInMessage in _fanInMessages)
             {
-                scopes.Push(
-                    SampleHelpers.CreateScopeWithPropagation(
-                        "kafka.consume",
-                        fanInMessage.Message.Headers,
-                        ConsumerBase.ExtractValues));
-            }
-            
-            Console.WriteLine($"Producing to {topic2} x2");
-            Producer.Produce(topic2, numMessages: 2, config, handleDelivery: true, isTombstone: false, explicitPartitions: true);
-            foreach (var scope in scopes)
-            {
-                scope.Dispose();
+                using (SampleHelpers.CreateScopeWithPropagation(
+                           "kafka.consume",
+                           fanInMessage.Message.Headers,
+                           ConsumerBase.ExtractValues))
+                {
+                    Console.WriteLine($"Producing to {topic2} - {iteration++} of 3");
+                    Producer.Produce(topic2, numMessages: 1, config, handleDelivery: true, isTombstone: false, explicitPartitions: true);
+                }
```

Comment:
This isn't really a fan-in scenario though right, this is more like "batched" handling 🤔 I would expect fan-in to "aggregate" the messages into one or more separate messages.

What was the reason for changing this test?

Reply:
I can leave this change out, it's not really "tied" with the actual code change. I changed it because the reason the `SpanContextExtractor` exists is actually for batch handling. If you're fanning in and aggregating, this scenario cannot be handled properly by DSM anyway. Maybe it'd make sense to rename the test ?

I actually wanted the consumed messages to have different pathways, but it's pretty hard to achieve, from the way pathway hashes are computed.

Reply:
Yeah, I think that's the difference really, the fact that now fan-in sort of "just works" (mostly because it's not really tracking what I _thought_ it was, as we discussed!). I don't mind leaving it in, but would prefer changing the naming (test and scenario etc) from "fan-in" to "batch". Not critical though, could definitely be separate PR to avoid hitting CI again